### PR TITLE
remove tx.origin checks

### PR DESCRIPTION
**Pull request by ccashwell**
Changes:
- Removed `tx.origin` validation
- Removed origin mismatch revert test
  - Incidentally, the claim that was being passed in for that test fails anyway, so that was replaced as well.

### CODE REVIEW

File: test/QuestFactory.t.sol [hunk 1]
```diff
@@ -306,8 +305,7 @@ contract TestQuestFactory is Test, Errors, Events, TestUtils {
     }
 
     function test_claimCompressed_revert_txOriginMismatch() public{
```
Comment (Quazia) [inline test/QuestFactory.t.sol:307]: Should we just remove the whole test?

**Review by Quazia — APPROVED** [review]
LGTM - I'd remove the whole test but not mission critical I reckon.

File: test/QuestFactory.t.sol (hunk 1 above)
Comment (ccashwell) [reply test/QuestFactory.t.sol]: Yeah, I always feel iffy about deleting tests that assert specific behavior like this. Open to removing it.

Comment (Quazia) [reply test/QuestFactory.t.sol]: Well we arn't asserting that behavior anymore, right?

**Pull request merged**